src/js/BookReader/Mode1Up.js

```diff
@@ -12,144 +12,219 @@ export class Mode1Up {
   constructor(br, bookModel) {
     this.br = br;
     this.book = bookModel;
+
+    /** @private */
+    this.$documentContainer = $('<div class="BRpageview" />');
+    /** @private */
+    this.screenDPI = calcScreenDPI();
+    /** @private */
+    this.LEAF_SPACING_IN = 0.2;
+
+    /**
+     * How much smaller the picture on screen is than the real-world item
+     *
+     * Mode1Up doesn't use the br.reduce because it is DPI aware. The reduction factor
+     * of a given leaf can change (since leaves can have different DPIs), but the real-world
+     * reduction is constant throughout.
+     */
+    this.realWorldReduce = 1;
   }
 
+  /** @private */
+  get $scrollContainer() { return this.br.refs.$brContainer; }
+
   /**
    * This is called when we switch to one page view
    */
   prepare() {
     const startLeaf = this.br.currentIndex();
 
-    this.br.refs.$brContainer.empty();
-    this.br.refs.$brContainer.css({
-      overflowY: 'scroll',
-      overflowX: 'auto'
-    });
-
-    this.br.refs.$brPageViewEl = $("<div class='BRpageview'></div>");
-    this.br.refs.$brContainer.append(this.br.refs.$brPageViewEl);
+    this.$scrollContainer
+      .empty()
+      .css({ overflowX: 'auto', overflowY: 'scroll' })
+      .append(this.$documentContainer);
 
     // Attaches to first child - child must be present
-    this.br.refs.$brContainer.dragscrollable();
-    this.br.bindGestures(this.br.refs.$brContainer);
-
-    // $$$ keep select enabled for now since disabling it breaks keyboard
-    //     nav in FF 3.6 (https://bugs.edge.launchpad.net/bookreader/+bug/544666)
-    // utils.disableSelect(this.$('#BRpageview'));
+    this.$scrollContainer.dragscrollable();
+    this.br.bindGestures(this.$scrollContainer);
 
+    // This calls drawLeafs
     this.resizePageView();
+
     this.br.jumpToIndex(startLeaf);
     this.br.updateBrClasses();
   }
 
-  drawLeafs() {
-    const { book } = this;
-    const containerHeight = this.br.refs.$brContainer.height();
-    const containerWidth = this.br.refs.$brPageViewEl.width();
-    const scrollTop = this.br.refs.$brContainer.prop('scrollTop');
-    const scrollBottom = scrollTop + containerHeight;
+  /**
+   * Get the number of pixels required to display the given inches with the given reduce
+   * @param {number} inches
```

Comment:
suggestion: update jsdoc to reflect args & return val

Reply:
Added description, but specifically not adding the types here. The inference is smart enough to figure these types out, so specifying them just means they're more likely to fall out of sync.